### PR TITLE
Update to ORC 1.7.2

### PR DESCRIPTION
**Pull request by homar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -800,6 +817,12 @@
                                         <exclude>overview.html</exclude>
                                     </excludes>
                                 </filter>
+                                <filter>
+                                    <artifact>org.apache.orc:orc-core</artifact>
+                                    <excludes>
+                                        <exclude>io/airlift/slice/**</exclude>
```
Comment (findepi) [inline pom.xml:823]: Should this go as an `<exclusion>`?

File: pom.xml [hunk 2]
```diff
@@ -654,7 +671,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.3</version>
+                <version>3.1.1</version>
```
Comment (findepi) [inline pom.xml:674]: related? separate commit?

File: pom.xml [hunk 3]
```diff
@@ -549,6 +549,23 @@
...
+            <artifactId>orc-core</artifactId>
+            <version>1.7.2</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>javax.xml.bind</groupId>
+                    <artifactId>jaxb-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jetbrains</groupId>
+                    <artifactId>annotations</artifactId>
```
Comment (findepi) [inline pom.xml:564]: Can you please document why this are excluded?

File: pom.xml [hunk 4]
```diff
@@ -549,6 +549,23 @@
             </exclusions>
         </dependency>
 
+        <!-- use newer ORC version -->
+        <dependency>
+            <groupId>org.apache.orc</groupId>
+            <artifactId>orc-core</artifactId>
+            <version>1.7.2</version>
```
Comment (findepi) [inline pom.xml:556]: What version was used here before?
Does hive automatically work with newer ORC version, or is any change needed within the hive code?

We generally don't use hive code to deal with ORC, but we eg use it when creating empty bucket files.
See `io.trino.plugin.hive.HiveMetadata#createEmptyFiles`.

**Review by electrum — CHANGES_REQUESTED** [review]
This seems to be a copy of #11, minus the exclusion for `org.threeten:threeten-extra`, which I'm not sure is intentional or an error. That original PR was not merged because the newer ORC version is not runtime compatible with Hive.

Have you tried running the Trino tests against this version?

**Comment by electrum** [comment]
The exclusion for `org.threeten:threeten-extra` is indeed needed. Without it, the library is included in the JAR, but is not relocated. It needs to be excluded entirely if not needed, excluded from relocation, or relocated.

**Comment by electrum** [comment]
I just published SNAPSHOT `3.1.2-19-20220302.022206-1` if you'd like to test out Trino against this PR.

File: pom.xml (hunk 1 above)
Comment (homar) [reply pom.xml:823]: > I just published SNAPSHOT `3.1.2-19-20220302.022206-1` if you'd like to test out Trino against this PR.

Thanks, will do !

**Comment by homar** [comment]
Thanks for the input Guys. This is not a way to go unfortunately

**Pull request closed without merging**